tests/testthat/test-parse_exclusions.R

```diff
@@ -0,0 +1,179 @@
+old_ops <- options(lintr.exclude = "#TeSt_NoLiNt",
+                   lintr.exclude_start = "#TeSt_NoLiNt_StArT",
+                   lintr.exclude_end = "#TeSt_NoLiNt_EnD")
+
+test_that("it returns an empty list if there are no exclusions", {
+  read_settings(NULL)
+  t1 <- tempfile()
+  on.exit(unlink(t1))
+  writeLines(
+    c("this",
+      "is",
+      "a",
+      "test"), t1)
+  expect_equal(parse_exclusions(t1), list())
+})
+
+test_that("it returns the line if one line is excluded", {
+  read_settings(NULL)
+
+  t1 <- tempfile()
+  on.exit(unlink(t1))
+  writeLines(
+    c("this",
+      "is #TeSt_NoLiNt",
+      "a",
+      "test"), t1)
+  expect_equal(parse_exclusions(t1), list(2))
+
+  t2 <- tempfile()
+  on.exit(unlink(t2))
+  writeLines(
+    c("this",
+      "is #TeSt_NoLiNt",
+      "a",
+      "test #TeSt_NoLiNt"), t2)
+  expect_equal(parse_exclusions(t2), list(c(2, 4)))
+})
+
+test_that("it supports specific linter exclusions", {
+  read_settings(NULL)
+
+  t1 <- tempfile()
+  on.exit(unlink(t1))
+  writeLines(
+    c("this",
+      "is #TeSt_NoLiNt: my_linter.",
+      "a",
+      "test"), t1)
+  expect_equal(parse_exclusions(t1), list(my_linter = 2))
+
+  t2 <- tempfile()
+  on.exit(unlink(t2))
+  writeLines(
+    c("this",
+      "is #TeSt_NoLiNt: my_linter.",
+      "a",
+      "test #TeSt_NoLiNt: my_linter2."), t2)
+  expect_equal(parse_exclusions(t2), list(my_linter = 2, my_linter2 = 4))
+
+  t3 <- tempfile()
+  on.exit(unlink(t3))
+  writeLines(
+    c("this",
+      "is #TeSt_NoLiNt: my_linter.",
+      "another",
+      "useful #TeSt_NoLiNt: my_linter.",
+      "test",
+      "testing #TeSt_NoLiNt: my_linter2."), t2)
+  expect_equal(parse_exclusions(t2), list(my_linter = c(2, 4), my_linter2 = 6))
+})
+
+test_that("it supports multiple linter exclusions", {
```

Comment:
CMIIW but do we need a test for

```
# nolint some_linter_that_doesn't_exist.
```

(which I guess should be a `warning`)

Reply:
That sounds useful, but how would we check it?
I don't think we can access `names(linters)` from within `parse_exclusions`...

Maybe add as an argument to `exclude` and check there?

Reply:
I see... for initial implementation then, let's leave it silently ignored for simplicity. One thing at a time :smile: